src/main/cluster.ts

```diff
@@ -643,17 +648,30 @@ export class Cluster implements ClusterModel, ClusterState {
       if (!this.allowedNamespaces.length) {
         return [];
       }
-      const resourceAccessStatuses = await Promise.all(
-        apiResources.map(apiResource => this.canI({
-          resource: apiResource.resource,
-          group: apiResource.group,
-          verb: "list",
-          namespace: this.allowedNamespaces[0]
-        }))
-      );
+      const resources = apiResources.filter((resource) => this.resourceAccessStatuses.get(resource) === undefined);
```

Comment:
Why to filter here if we check it below with `!this.resourceAccessStatuses.get(apiResource)`?

Reply:
This filter returns only resources which don't have any result. Later we check only falsey values.

Reply:
I think that is probably needless complexity since the check below will just succeed quickly.

Reply:
> I think that is probably needless complexity since the check below will just succeed quickly.

Agreed, it might be unnecessary optimisation but should not hurt either. We are probably going to improve this in follow-up PRs where we have new possibilities for cleanup.